### PR TITLE
PP-5300 Introduce Direct Debit Specific GET Payments Endpoint

### PR DESCRIPTION
**Pull request by RoryMalcolm**
- Migrates logic for Get Payments endpoint for Direct Debit payments
- Changes return type to match DD Payments information

### CODE REVIEW

File: src/main/java/uk/gov/pay/api/resources/PaymentsResource.java [hunk 1]
```diff
@@ -115,6 +120,28 @@ public Response getPayment(@ApiParam(value = "accountId", hidden = true) @Auth A
...
+                                          @PathParam("paymentId")
+                                          @ApiParam(name = "paymentId",
+                                                  value = "Payment identifier", 
+                                                  example = "hu20sqlact5260q2nanm0q8u93")
+                                                  String paymentId) {
+        
+        logger.info("Direct Debit Payment request - paymentId={}", paymentId);
+        DirectDebitPayment payment = getDirectDebitPaymentService.getDirectDebitPayment(account, paymentId);
+        return Response.ok(payment).build();
+    }
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/api/resources/PaymentsResource.java]: This should be in DirectDebitPaymentsResource

File: src/main/java/uk/gov/pay/api/service/GetDirectDebitPaymentService.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        this.connectorUriGenerator = connectorUriGenerator;
+    }
+
+    public DirectDebitPayment getDirectDebitPayment(Account account, String paymentId) {
+        Response connectorResponse = client
+                .target(connectorUriGenerator.chargeURI(account, paymentId))
+                .request()
+                .get();
+        if (connectorResponse.getStatus() == SC_OK) {
+            DirectDebitConnectorCreatePaymentResponse ddConnectorResponse =
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/api/service/GetDirectDebitPaymentService.java]: Could you rename `DirectDebitConnectorCreatePaymentResponse` to remove "Create" as it's now used for both create and get

File: src/main/java/uk/gov/pay/api/service/GetDirectDebitPaymentService.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                    .withPaymentProvider(ddConnectorResponse.getPaymentProvider())
+                    .withState(ddConnectorResponse.getState())
+                    .withCreatedDate(ddConnectorResponse.getCreatedDate())
+                    .withReference(ddConnectorResponse.getReference())
+                    .withSelfLink(publicApiUriGenerator.getDirectDebitPaymentURI(paymentId))
+                    .withMandateLink(publicApiUriGenerator.getMandateURI(ddConnectorResponse.getMandateId()))
+                    .withEventsLink(publicApiUriGenerator.getMandateEventsURI(ddConnectorResponse.getMandateId()))
+                    .build();
+        }
+        throw new GetChargeException(connectorResponse);
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/api/service/GetDirectDebitPaymentService.java]: As a style thing I'd prefer to invert the if and throw the exception earlier up, just to reduce nesting

File: src/main/java/uk/gov/pay/api/service/PublicApiUriGenerator.java [hunk 4]
```diff
@@ -77,7 +77,13 @@ public URI getDirectDebitMandateURI(String mandateId) {
...
                 .build(mandateId);
     }
-
+    
+    public URI getDirectDebitPaymentSelfURI(String paymentId) {
+        return UriBuilder.fromUri(baseUrl)
+                .path("/v1/directdebit/payments/{paymentId}")
+                .build(paymentId);
+    }
+
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/api/service/PublicApiUriGenerator.java]: This is the same as `getDirectDebitPaymentURI` - just use that method

File: swagger/swagger.json [hunk 5]
```diff
@@ -11,6 +11,30 @@
   } ],
```
Comment (stephencdaly) [inline swagger/swagger.json]: Don't commit the swagger for now. The new endpoint shouldn't show up in it when you move it. We'll have a later story to add to the swagger file.

**Comment by RoryMalcolm** [comment]
@stephencdaly Made the changes

File: src/main/java/uk/gov/pay/api/service/GetDirectDebitPaymentService.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+                    .withAmount(ddConnectorResponse.getAmount())
+                    .withPaymentId(ddConnectorResponse.getPaymentExternalId())
+                    .withPaymentProvider(ddConnectorResponse.getPaymentProvider())
+                    .withState(ddConnectorResponse.getState())
+                    .withCreatedDate(ddConnectorResponse.getCreatedDate())
+                    .withReference(ddConnectorResponse.getReference())
+                    .withSelfLink(publicApiUriGenerator.getDirectDebitPaymentURI(paymentId))
+                    .withMandateLink(publicApiUriGenerator.getMandateURI(ddConnectorResponse.getMandateId()))
+                    .withEventsLink(publicApiUriGenerator.getMandateEventsURI(ddConnectorResponse.getMandateId()))
+                    .build();
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/api/service/GetDirectDebitPaymentService.java]: Because we now do this in 2 places, I reckon it's probably worthwhile to have a `from(DirectDebitConnectorPaymentResponse paymentResponse)` method on `DirectDebitPayment` to avoid the duplication.
You can do this in another PR if you prefer

File: swagger/swagger.json [hunk 7]
```diff
@@ -11,6 +11,30 @@
...
   "schemes" : [ "https" ],
   "paths" : {
+    "/v1/directdebit/payments/{paymentId}" : {
```
Comment (stephencdaly) [inline swagger/swagger.json]: Remove the changes to the swagger file from the commit
